File: business/business-oj/src/main/java/com/glowxq/oj/topic/enums/TopicType.java
package com.glowxq.oj.topic.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Arrays;
import java.util.List;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum TopicType implements BaseType {

    // 其它
    Other("Other", "其它", "除比赛外的种类"),

    // 测试
    Exercise("Exercise", "练习", "默认设置 无限制的随意测试"),

    // 作业
    Homework("Homework", "作业", "默认设置 明确提交时间,过期可以补交但是扣分"),

    // 比赛
    Contest("Contest", "比赛", "默认设置 过期禁止提交"),
    ;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 解释
     */
    private final String description;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static TopicType matchCode(String code) {
        for (TopicType pushStatus : TopicType.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }

    public List<String> queryListCode() {
        if (Other.equals(this)) {
            return Arrays.stream(values()).filter(value -> !Contest.equals(value)).map(TopicType::getCode).toList();
        }
        return List.of(this.getCode());
    }
}

File: oj-web/src/enums/oj/topic/TopicType.ts
import type { BaseEnum } from '@/enums/base';

// 主题类型枚举接口
interface TopicTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 主题类型枚举
 * 用于标识不同的主题类型
 */
export const TopicType: { [key: string]: TopicTypeEnum } = {
  /**
   * 练习
   */
  EXERCISE: {
    code: 'Exercise',
    name: '练习',
    tooltip: '默认设置 无限制的随意测试',
    text: '默认设置 无限制的随意测试'
  },

  /**
   * 作业
   */
  HOMEWORK: {
    code: 'Homework',
    name: '作业',
    tooltip: '默认设置 明确提交时间,过期可以补交但是扣分',
    text: '默认设置 明确提交时间,过期可以补交但是扣分'
  },

  /**
   * 比赛
   */
  CONTEST: {
    code: 'Contest',
    name: '比赛',
    tooltip: '默认设置 过期禁止提交',
    text: '默认设置 过期禁止提交'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): TopicTypeEnum | null {
  return Object.values(TopicType).find(item => item.code === code) || null;
} 